internal/backup/volumesnapshotcontent_action.go

```diff
@@ -73,9 +74,26 @@ func (p *VolumeSnapshotContentBackupItemAction) Execute(item runtime.Unstructure
 
 	snapContMap, err := runtime.DefaultUnstructuredConverter.ToUnstructured(&snapCont)
 	if err != nil {
-		return nil, nil, errors.WithStack(err)
+		return nil, nil, "", nil, errors.WithStack(err)
 	}
 
 	p.Log.Infof("Returning from VolumeSnapshotContentBackupItemAction with %d additionalItems to backup", len(additionalItems))
-	return &unstructured.Unstructured{Object: snapContMap}, additionalItems, nil
+	return &unstructured.Unstructured{Object: snapContMap}, additionalItems, "", nil, nil
```

Comment:
Do we need to add the VSC object to itemToUpdate as well? This is decided by two factors:
1. Whether the VSC object is modified during the async operation -- wait for it becomes readyToUse?  (I think the answer is yes)
2. Whether the changes in 1 is critical to Velero so they must be persisted

Reply:
https://github.com/kubernetes-csi/external-snapshotter/blob/master/client/apis/volumesnapshot/v1/types.go#L121-L130
Per the external-snapshotter comments, the `ReadyToUse` in VolumeSnapshotStatus and VolumeSnapshotContentStauts should be the same, and fields in VolumeSnapshotStatus are updated based on fields in VolumeSnapshotContentStatus.
I think just watching over VolumeSnapshotContentStauts's change should be enough.

Reply:
Make one item as itemToUpdate means the item will be update to backup tarball the second time after all async operation finishes.
Or in another words, if we don't need to save the up-to-date object in backup tarball, we don't even need to make VS object as an itemToUpdate either.

Reply:
@Lyndon-Li The volumesnapshot will be in the tarball in any case because the PVC plugin returns it as an additional item. It is the *current item* handled by this plugin, which means that after this plugin execute exits, Velero adds this item to the backuup. Adding it to itemsToUpdate just means that we get an updated copy of this resource during finalize. If we don't add it to itemsToUpdate, then the tarball will have the not-yet-ready version.

@shubham-pampattiwar the current OADP datamover code requires the VS and VSC to be in the tarball on restore, right? Let me know if I'm misremembering.

Reply:
@sseago 
Exactly, by that I meant, we need to put the VSC object into itemToUpdate as same as VS object

Reply:
I see. I will add the VSC in the itemsToUpdate to make sure its state is updated after the operation completes.